### PR TITLE
Drop proxy binary messenger

### PR DESCRIPTION
**Pull request by maios**
### What does this pull request do?
 Drop `ProxyBinaryMessenger`, instead, setup channel with a messageChannelSuffix which is now supported by Pigeon

### Pull request checklist:
 - [ ] Add a changelog entry.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.

### CODE REVIEW

File: ios/Classes/EventHandler.swift
Comment (evil159) [inline ios/Classes/EventHandler.swift:48]: Channel suffix should be a required parameter here.

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/snapshotter/SnapshotterInstanceManager.kt
Comment (evil159) [inline android/src/main/kotlin/com/mapbox/maps/mapbox_maps/snapshotter/SnapshotterInstanceManager.kt:35]: Here should be a non-static suffix

File: lib/src/offline/offline_manager.dart
Comment (evil159) [inline lib/src/offline/offline_manager.dart:1]: I think this was an issue before as well - `EventChannel` created in `loadStylePack` does not have any suffixes - meaning it will broadcast to any listeners loading the same style uri, better attach some suffix to it as well.

File: lib/src/offline/tile_store.dart
Comment (evil159) [inline lib/src/offline/tile_store.dart:1]: Here in `loadTileRegion` we create `EventChannel` with an id of the region, I think in case of multiple tile store instances those ids could clash, safer to add a suffix there as well.

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/EventHandler.kt
Comment (evil159) [inline android/src/main/kotlin/com/mapbox/maps/mapbox_maps/EventHandler.kt:41]: This should be a required parameter.

File: lib/src/mapbox_maps_platform.dart
Comment (evil159) [inline lib/src/mapbox_maps_platform.dart:6]: We have to add suffix to the name of this channel otherwise all kinds of events could end up broadcasting across all active map instances.

File: lib/src/snapshotter/snapshotter.dart
Comment (evil159) [inline lib/src/snapshotter/snapshotter.dart:49]: I noticed a pattern where binary messenger is specified explicitly, for consistency it should be specified here as well.

File: lib/src/map_events.dart
Comment (evil159) [inline lib/src/map_events.dart:43]: Additionally binary messenger should be a required parameter as well, since in all other places binary messenger is specified explicitly, having it as a requirement will help with consistency.

**Review by evil159 — CHANGES_REQUESTED** [review]
There some aspects to improve, mainly when it comes to multi-instance scenarios, but looking good overall.

File: lib/src/annotation/polyline_annotation_manager.dart
Comment (evil159) [reply lib/src/annotation/polyline_annotation_manager.dart:12]: I setup linter to produce an error for cases like this(private types in public API), but it seems that those errors don't fail the build, I guess we'll have check it on CI in order to catch them.

Comment (evil159) [reply lib/src/annotation/polyline_annotation_manager.dart:12]: Here's the PR https://github.com/mapbox/mapbox-maps-flutter/pull/735

File: lib/src/snapshotter/snapshotter.dart
Comment (maios) [reply lib/src/snapshotter/snapshotter.dart:49]: I don't think it is necessary, as stated by the doc, so we can leave it as optional null
> The [binaryMessenger] named argument is available for dependency injection.  If it is left null, the default BinaryMessenger will be used which routes to the host platform.

File: ios/Classes/MapboxMapFactory.swift
Comment (maios) [reply ios/Classes/MapboxMapFactory.swift:58]: it's the same as before 😅 this one is only clean up the code a bit

**Review by pjleonard37 — APPROVED** [review]
Looks good from my side. I think ready to go once we get @evil159 's approval

File: lib/src/annotation/annotation_manager.dart
Comment (evil159) [inline lib/src/annotation/annotation_manager.dart:81]: nit: `channelSuffix` is not stored here anymore, might as well be removed from here. Subclasses might as well declare their own constructor parameter: `required super.channelSuffix` > `required String channelSuffix`)

File: lib/src/mapbox_maps_platform.dart
Comment (evil159) [inline lib/src/mapbox_maps_platform.dart:7]: There should be suffix here, otherwise this method channel will broadcast all events to all active map instances.

**Review by evil159 — COMMENTED** [review]
@maios thank you for addressing the feedback. I think we are almost there, I have left a couple of nits(in my opinion the binary messenger parameter should be removed), and I noticed there is one method channel in `_MapboxMapsPlatform` lacking the channel suffix.
Otherwise, it's looking very good!

P.S. This is an internal change, but its impact is broad. Should we note it in the changelog?

**Review by evil159 — APPROVED** [review]
Nicely done!

**Comment by pjleonard37** [comment]
@maios  -- I think the `build-android` job is failing because of this unresolved reference to $messageChannel. 

https://github.com/mapbox/mapbox-maps-flutter/pull/715/files#diff-f72e4d76e69116de99f71b5e16fe8b73874d764869c822e44a81453325f7234dR100

Should this be `$channelSuffix`?

**Review by pjleonard37 — APPROVED** [review]

**Pull request merged**
